webapp/src/js/enketo/widgets/dynamic-url.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const Widget = require( 'enketo-core/src/js/widget' ).default;
+const $ = require( 'jquery' );
+require( 'enketo-core/src/js/plugins' );
+
+/**
+ * Supports dynamically generated clickable links in the form markdown.
+ * https://github.com/medic/cht-core/issues/3349
+ */
+class DynamicUrlWidget extends Widget {
+  static get selector() {
+    return 'a.dynamic-url';
+  }
+
+  _init() {
+    const urlElement = $( this.element ).find('.url');
+    $( this.element ).click(() => window.open(urlElement.text(), '_blank'));
```

Comment:
So, running in WebView on Android it pops a new tab in my default browser (as expected).  When running in xWalk on Android, it also pops a new tab in my default browser, but when I switch back to the CHT Android app, I see that the Confirmation dialog has been opened telling me "This form is not finished...Are you sure you want to leave?".  If I hit "No", it just takes me back to my form, but if I hit "Yes" is closes the form and takes me back out to the root page.

Going to do some digging to see if I can figure out how to get it to not pop the dialog...

Reply:
Okay, so I ended up switching up the code here so that instead of dynamically opening the link via a click-listener, I am just dynamically updating the actual href on the anchor when the value in the `.url` span changes.